js/components/frame.js

```diff
@@ -877,10 +887,27 @@ class Frame extends ImmutableComponent {
           method = (currentDetail, originalDetail) =>
             windowActions.setAutofillCreditCardDetail(currentDetail, originalDetail)
           break
+        case messages.TORRENT_MESSAGE:
+          // Relay torrent IPC from the webview to a browser process
+          method = (message) => {
+            if (typeof message.clientKey !== 'string') {
+              throw new Error('Invalid or missing clientKey: ' + JSON.stringify(message))
+            }
+            this.torrentChannelID = message.clientKey
+            ipc.send(messages.TORRENT_MESSAGE, message)
+          }
+          break
       }
       method.apply(this, e.args)
     })
 
+    // Relay torrent IPC from the browser process back to the webview
+    ipc.on(messages.TORRENT_MESSAGE, (e, message) => {
```

Comment:
this  may not be necessary either; i think the browser process can just do `event.sender.send` after it gets a message from the webview in `app/browser/webtorrent.js`